src/aleph/vm/orchestrator/tasks.py

```diff
@@ -127,3 +134,46 @@ async def stop_watch_for_messages_task(app: web.Application):
         await app["messages_listener"]
     except asyncio.CancelledError:
         logger.debug("Task messages_listener is cancelled now")
+
+
+async def monitor_payments(app: web.Application):
+    logger.debug("Monitoring balances")
+    pool: VmPool = app["vm_pool"]
+    while True:
+        await asyncio.sleep(settings.PAYMENT_MONITOR_INTERVAL)
+
+        # Check if the balance held in the wallet is sufficient holder tier resources
+        for sender, executions in pool.get_executions_by_sender(payment=PaymentType.hold):
```

Comment:
Should we explicitly sort them chronologically before checking them? What should be the default behavior when looking for the first VMs to shut down? Latest first?